rb_link/src/publisher.rs

```diff
@@ -0,0 +1,83 @@
+use crate::server::*;
+use std::time::Instant;
+
+/// The Subscriber can be used to listen to messages from specific probes
+pub trait Subscriber {
+    /// called when a new cycle's messages arrived,
+    /// takes the Subscribe probes's message sent in the cycle
+    /// the return value will be put into the server
+    fn update(&mut self, messages: Vec<B2RMessage>) -> Vec<R2BMessage>;
+    /// Specify the ID of the subscribed probe.
+    fn subscribed_ids(&self) -> Vec<u32>;
+}
+
+/// A wrapper for B2RServer that to used the server and getter in event driven style
+pub struct B2RPublisher {
+    server: B2RServer,
+    cycle_getter: CycleGetter,
+    subscribers: Vec<Box<dyn Subscriber>>,
+}
+
+impl B2RPublisher {
+    /// Creates a new B2RPublisher instance with the specified socket path.
+    pub fn new_with(path: &str) -> Self {
```

Comment:
How does this module works with other modules? It will take the unix stream. What if I want to use b2r_cache and r2b_cache while also using Subscriber framework?
As a framework, different part should be able to be used together, not splited.